### PR TITLE
fix: comments in #24

### PR DESCRIPTION
**Pull request by amirlazarovich**

### CODE REVIEW

File: android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt [hunk 1]
```diff
@@ -15,8 +15,12 @@ import com.gdelataillade.alarm.features.VolumeHandler
 import com.gdelataillade.alarm.utils.nextDateInMillis
 import io.flutter.Log
 import io.flutter.plugin.common.MethodChannel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
 
 class AlarmService : Service() {
+    private val serviceScope = CoroutineScope(Dispatchers.Main + SupervisorJob())
```
Comment (maciekczwa) [inline android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt]: we never cancel the scope.
We should make something like this:
https://stackoverflow.com/a/63407811/1430242

Use Dispatchers.IO

Just call from service ```serviceScope.launch { audioHandler.playAudio... }```
AudioHandler shouldn't know about the scope.

If we need the context in playAudio we can mark it as suspend

We can also just move the scope to AudioHandler.

File: android/src/main/kotlin/com/gdelataillade/alarm/features/AudioHandler.kt [hunk 2]
```diff
@@ -25,52 +30,66 @@ class AudioHandler(private val context: Context) {
...
-                if (fadeDuration != null && fadeDuration > 0) {
-                    val timer = Timer(true)
-                    timers[id] = timer
-                    startFadeIn(this, fadeDuration, timer)
+                    if (fadeDuration != null && fadeDuration > 0) {
+                        // Timer operations should also be offloaded
+                        scope.launch(Dispatchers.Default) {
+                            val timer = Timer(true)
+                            timers[id] = timer
+                            startFadeIn(this@apply, fadeDuration, timer)
```
Comment (maciekczwa) [inline android/src/main/kotlin/com/gdelataillade/alarm/features/AudioHandler.kt]: If we are using coroutines, we can get rid of these timers. Just simple loop with delay inside suspend method.

File: android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt (hunk 1 above)
Comment (amirlazarovich) [reply android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt]: done

File: android/src/main/kotlin/com/gdelataillade/alarm/features/AudioHandler.kt (hunk 2 above)
Comment (amirlazarovich) [reply android/src/main/kotlin/com/gdelataillade/alarm/features/AudioHandler.kt]: done

**Review by maciekczwa — APPROVED** [review]

**Pull request merged**